test/jellyfish/config_reader_test.exs

```diff
@@ -129,37 +129,52 @@ defmodule Jellyfish.ConfigReaderTest do
   end
 
   test "read_dist_config/0 NODES_LIST" do
-    with_env ["JF_DIST_ENABLED", "JF_DIST_COOKIE", "JF_DIST_NODE_NAME", "JF_DIST_NODES"] do
+    with_env [
+      "JF_DIST_ENABLED",
+      "JF_DIST_MODE",
+      "JF_DIST_COOKIE",
+      "JF_DIST_NODE_NAME",
+      "JF_DIST_NODES"
+    ] do
+      {:ok, hostname} = :inet.gethostname()
+
       assert ConfigReader.read_dist_config() == [
                enabled: false,
+               mode: nil,
                strategy: nil,
                node_name: nil,
                cookie: nil,
                strategy_config: nil
              ]
 
       System.put_env("JF_DIST_ENABLED", "true")
-      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
-      System.put_env("JF_DIST_COOKIE", "testcookie")
-      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
-      System.put_env("JF_DIST_NODE_NAME", "testnodename@abc@def")
-      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
-      System.put_env("JF_DIST_NODE_NAME", "testnodename")
-      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
-      System.put_env("JF_DIST_NODE_NAME", "testnodename@127.0.0.1")
 
       assert ConfigReader.read_dist_config() == [
                enabled: true,
+               mode: :shortnames,
                strategy: Cluster.Strategy.Epmd,
-               node_name: :"testnodename@127.0.0.1",
-               cookie: :testcookie,
+               node_name: :"jellyfish@#{hostname}",
+               cookie: :jellyfish_cookie,
                strategy_config: [hosts: []]
              ]
 
+      System.put_env("JF_DIST_NODE_NAME", "testnodename@abc@def")
+      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
+      System.put_env("JF_DIST_NODE_NAME", "testnodename")
+      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
+      System.delete_env("JF_DIST_NODE_NAME")
+      assert ConfigReader.read_dist_config()
+      System.put_env("JF_DIST_MODE", "invalid")
+      assert_raise RuntimeError, fn -> ConfigReader.read_dist_config() end
+
+      System.put_env("JF_DIST_MODE", "name")
+      System.put_env("JF_DIST_COOKIE", "testcookie")
+      System.put_env("JF_DIST_NODE_NAME", "testnodename@127.0.0.1")
```

Comment:
We should also check the situation in which we applied some of the required ENVs. Currently, this test checks only when we don't have set any ENVs and then that we set all of them.

Reply:
Yeap as right now there are no required env vars :stuck_out_tongue: